model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/CacheManager.java

```diff
@@ -206,28 +256,39 @@ private Iterator<Map.Entry<String, Revisioned>> getEntryIterator(Predicate<Map.E
     }
 
 
-    public void sendInvalidationEvents(KeycloakSession session, Collection<InvalidationEvent> invalidationEvents, String eventKey) {
+    public void sendInvalidationEvents(KeycloakSession session, Collection<InvalidationEvent> invalidationEvents, String eventKey, InvalidationManager invalidationManager) {
         ClusterProvider clusterProvider = session.getProvider(ClusterProvider.class);
 
         // Maybe add InvalidationEvent, which will be collection of all invalidationEvents? That will reduce cluster traffic even more.
         for (InvalidationEvent event : invalidationEvents) {
             clusterProvider.notify(eventKey, event, true, ClusterProvider.DCNotify.ALL_DCS);
         }
+        if (invalidationManager != null) {
+            for (InvalidationEvent event : invalidationManager.getInvalidationEvents()) {
+                clusterProvider.notify(eventKey, event, true, ClusterProvider.DCNotify.ALL_DCS);
+            }
+        }
     }
 
 
     public void invalidationEventReceived(InvalidationEvent event) {
-        Set<String> invalidations = new HashSet<>();
+        InvalidationManager invalidationManager = new InvalidationManager();
 
-        addInvalidationsFromEvent(event, invalidations);
+        addInvalidationsFromEvent(event, invalidationManager.getModelInvalidations(), invalidationManager);
 
-        getLogger().debugf("[%s] Invalidating %d cache items after received event %s", cache.getCacheManager().getAddress(), invalidations.size(), event);
+        getLogger().debugf("[%s] Invalidating %d cache items after received event %s", cache.getCacheManager().getAddress(), invalidationManager.getModelInvalidations().size(), event);
 
-        for (String invalidation : invalidations) {
-            invalidateObject(invalidation);
-        }
+        runInvalidations(invalidationManager);
+    }
+
+    protected abstract void addInvalidationsFromEvent(InvalidationEvent event, Set<String> invalidations, InvalidationManager invalidationManager);
+
+    public Long getConfiguredIdleTime(String modelName) {
+        return cacheConfig.getLong(modelName + ".idle", -1L);
```

Comment:
using the model name to pull the configured idle or lifespan for that particular model